### ISSUE TITLE
Angular CLI 1.5.0-rc.6 does not support @angular/service-worker@5.0.0-rc.8

### ISSUE THREAD

**Issue by niikoo** [angular/angular-cli#8247]
### Bug Report or Feature Request (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.

**ng -v**

Angular CLI: 1.5.0-rc.6
Node: 8.5.0
OS: win32 x64
Angular: 5.0.0-rc.8
... animations, common, compiler, compiler-cli, core, forms
... http, platform-browser, platform-browser-dynamic
... platform-server, router

@angular/cdk: 2.0.0-beta.12
@angular/cli: 1.5.0-rc.6
@angular/material: 2.0.0-beta.12
@angular/service-worker: 5.0.0-rc.8
@angular-devkit/build-optimizer: 0.0.31
@angular-devkit/core: 0.0.20
@angular-devkit/schematics: 0.0.34
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.8.0-rc.6
@schematics/angular: 0.0.48
typescript: 2.5.3
webpack: 3.8.1

### Repro steps.

ng build --prod --aot --base-href=.

### The log given by the failure.

**Build fails with error: **

The installed version of @angular/service-worker is 5.0.0-rc.8. This version of the CLI
requires the @angular/service-worker version to satisfy >= 1.0.0-beta.5 < 2.0.0. Please upgrade
your service worker version.
Error: The installed version of @angular/service-worker is 5.0.0-rc.8. This version of the CLI
requires the @angular/service-worker version to satisfy >= 1.0.0-beta.5 < 2.0.0. Please upgrade
your service worker version.

### Desired functionality.

The newest @angular/cli isn't compatible with any of the new @angular/service-worker versions; which it should.

### Mention any other details that might be useful.

The old version of @angular/service-worker is working, 1.0.0-beta.16.

**Comment by avatsaev** [comment]
Same issue here

**Comment by cgatian** [comment]
Perhaps this is by design. Using 1.5 CLI and 5.0 @angular/service-worker also reports this error

**Comment by csentis** [comment]
Just looked at the latest commits and saw that service-worker got pinned @1.0.0-beta.16 for the 1.5.0 release.

A pity :)

**Comment by Splaktar** [comment]
Yep, `@angular/service-worker` support did not make it into Angular CLI `v1.5.0`.

**Comment by MattiJarvinen-BA** [comment]
Also documentation on `@angular/service-worker` is lacking. Even angular.io doesn't use it yet so no surprises here.

**Comment by mir4ef** [comment]
@Splaktar can we expect to see it in a 1.5.x patch release or the plan is to include it in a minor/major cli release?

**Comment by Splaktar** [comment]
I'm not an authority on this topic, but https://github.com/angular/angular-cli/issues/8224 seems to indicate that it would be part of `1.6.0`. Based on the rapid rate of development on the CLI, I don't think that will be a long wait.

**Comment by csentis** [comment]
Likewise #8080.

**Comment by Yeradon** [comment]
There is no built-in support for service worker in 1.5. But since the `@angular/service-worker` package has it’s cli separate you can use it with 1.5. There is a good explantation for this by @webmaxru [here](https://medium.com/@webmaxru/a-new-angular-service-worker-creating-automatic-progressive-web-apps-part-1-theory-37d7d7647cc7) and [here](https://medium.com/@webmaxru/a-new-angular-service-worker-creating-automatic-progressive-web-apps-part-2-practice-3221471269a1). Just make sure to remove the service-worker entry from the `.angular-cli.json`, because this will cause a fatal error.

**Comment by web-dave** [comment]
the buildin feature of ServiceWorker ist not updated.
To user ServiceWorker:

1. remove `"serviceWorker": true,` from `.angular-cli.json`

2. add these scripts to your `package.json`
```
    "build:prod": "ng build --prod",
    "ngsw-config": "node_modules/.bin/ngsw-config dist src/ngsw-config.json",
    "ngsw-copy": "cp node_modules/@angular/service-worker/ngsw-worker.js dist/",
    "postbuild:prod": "npm run ngsw-config && npm run ngsw-copy"
```
3.  run `npm run build:prod`

**Comment by johanchouquet** [comment]
I now have cli 1.6.0-rc.0 and confirm that without `"serviceWorker": true` in the `.angular-cli.json` file DOES resolve the issue.

**Comment by jwuliger** [comment]
Hello All ~

I am having the same issue even after trying what @johanchouquet said. I have `"serviceWorker": true` in my `.angular-cli.json` file yet I still get this error when trying to build:

```
The installed version of @angular/service-worker is 5.1.0-beta.2. This version of the CLI
requires the @angular/service-worker version to satisfy >= 1.0.0-beta.5 < 2.0.0. Please upgrade
your service worker version.
```

Here is my `ng -v` info:

```
Angular CLI: 1.6.0-rc.1-930437e
Node: 8.9.1
OS: win32 x64
Angular: 5.1.0-beta.2
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router, service-worker

@angular/cli: 1.6.0-rc.1-930437e
@angular-devkit/build-optimizer: 0.0.34
@angular-devkit/core: 0.0.22
@angular-devkit/schematics: 0.0.38
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.0-rc.1-930437e
@schematics/angular: 0.1.8
typescript: 2.6.2
webpack: 3.8.1
```

Please let me know if there is anyway around this. Thanks!

**Comment by chai-jay** [comment]
If you want to use @angular/service-worker with the Angular CLI <1.6.0, you will have to install the 1.0.0 Beta version of the service worker for now. 

So run `npm install --save @angular/service-worker@"~1.0.0-beta"` then you can keep you `"serviceWorker": true` config in your `.angular-cli.json`.

**Comment by cexbrayat** [comment]
I think there is still an issue with `1.6.0-rc.1`.

If you try to use `@angular/service-worker` in `5.1.0-rc.1` for example, you run into:
```
The installed version of @angular/service-worker is 5.1.0-rc.1. This version of the CLI
requires the @angular/service-worker version to satisfy >= 1.0.0-beta.5 < 2.0.0. Please upgrade
your service worker version.
```

whereas using `5.0.5` everything is fine.

**Comment by mackelito** [comment]
works in 1.6.0-rc.2

**Comment by cexbrayat** [comment]
@dalu this is fixed in the upcoming 1.6 
You can try 1.6.0-rc.2 if you want (first release with the fix).

**Comment by alan-agius4** [comment]
Closing as RC versions of version 1 is no longer supported.
